### PR TITLE
fix: Text Content does not propagate changes from Settings Panel if Previous Edit was on Canvas

### LINKED ISSUE #3021: Text Content does not propagate changes from Settings Panel if Previous Edit was on Canvas
image

Start Editing Text on Canvas then click to Settings/Text Content and proceed with editing. Text on canvas will not be updated until blur.

### PR DESCRIPTION
**Pull request by istarkov**
## Description

Closes #3021

The issue was that TextEditor when you switch on setting panel were active and in no way synchronized with state having it's uncontrollable nature.

Here we are disabling editing mode on editor blur.

## Steps for reproduction

1. click button
2. expect xyz

## Code Review

- [ ] hi @kof, I need you to do
  - conceptual review (architecture, feature-correctness)
  - detailed review (read every line)
  - test it on preview

## Before requesting a review

- [ ] made a self-review
- [ ] added inline comments where things may be not obvious (the "why", not "what")

## Before merging

- [ ] tested locally and on preview environment (preview dev login: 5de6)
- [ ] updated [test cases](https://github.com/webstudio-is/webstudio/blob/main/apps/builder/docs/test-cases.md) document
- [ ] added tests
- [ ] if any new env variables are added, added them to `.env.example` and the `builder/env-check.js` if mandatory

### CODE REVIEW

File: apps/builder/app/canvas/features/text-editor/text-editor.tsx [hunk 1]
```diff
@@ -30,6 +30,7 @@ import {
   $convertTextToLexical,
 } from "./interop";
 import { colord } from "colord";
+import { useEffectEvent } from "~/builder/features/ai/hooks/effect-event";
```
Comment (TrySound) [inline apps/builder/app/canvas/features/text-editor/text-editor.tsx:33]: need to move somewhere or use bare refs

**Review by TrySound — APPROVED** [review]

File: apps/builder/app/canvas/features/text-editor/text-editor.tsx [hunk 2]
```diff
@@ -63,6 +64,27 @@ const AutofocusPlugin = () => {
...
+    }
+
+    rootElement.addEventListener("blur", onBlur);
+
+    return () => {
+      rootElement.removeEventListener("blur", onBlur);
+    };
+  }, [editor, onBlur]);
+
+  return null;
```
Comment (kof) [inline apps/builder/app/canvas/features/text-editor/text-editor.tsx:85]: does it still require a null? react takes undefined these days
```suggestion
  return;
```

Comment (istarkov) [reply apps/builder/app/canvas/features/text-editor/text-editor.tsx:85]: image

**Review by kof — APPROVED** [review]

File: apps/builder/app/builder/features/ai/hooks/long-press-toggle.ts [hunk 3]
```diff
@@ -1,5 +1,5 @@
 import { type SyntheticEvent, type KeyboardEvent, useRef } from "react";
-import { useEffectEvent } from "./effect-event";
+import { useEffectEvent } from "../../../../shared/hook-utils/effect-event";
```
Comment (kof) [inline apps/builder/app/builder/features/ai/hooks/long-press-toggle.ts:2]: lets use shorter path ~/shared/hook-utils

File: apps/builder/app/builder/features/ai/ai-command-bar.tsx [hunk 4]
```diff
@@ -43,7 +43,7 @@ import { useLongPressToggle } from "./hooks/long-press-toggle";
 import { AiCommandBarButton } from "./ai-button";
 import { fetchTranscription } from "./ai-fetch-transcription";
 import { fetchResult } from "./ai-fetch-result";
-import { useEffectEvent } from "./hooks/effect-event";
+import { useEffectEvent } from "../../../shared/hook-utils/effect-event";
```
Comment (kof) [inline apps/builder/app/builder/features/ai/ai-command-bar.tsx:46]: same ~/shared/hook-utils

**Pull request merged**